### PR TITLE
Implement translation for `Regexp` node

### LINKED ISSUE #156: Implement translation for `PM_REGULAR_EXPRESSION_NODE`

### PR DESCRIPTION
**Pull request by egiurleo**
Closes #156.

### Test plan
See included automated tests.

### CODE REVIEW

File: parser/prism/Parser.cc [hunk 1]
```diff
@@ -24,4 +24,8 @@ std::string_view Parser::resolveConstant(pm_constant_id_t constant_id) {
     return std::string_view(reinterpret_cast<const char *>(constant->start), constant->length);
 }
 
+std::string_view Parser::extractString(pm_string_t *string) {
+    return std::string_view(reinterpret_cast<const char *>(pm_string_source(string)), pm_string_length(string));
```
Comment (amomchilov) [inline parser/prism/Parser.cc:28]: Let's add a check here, just to make sure we don't leak:

```suggestion
    if (string->type != PM_STRING_CONSTANT || string->type != PM_STRING_SHARED) {
        Exception::raise("This string needs to be freed eventually with `pm_string_free`, but that isn't implemented yet");
    }

    return std::string_view(reinterpret_cast<const char *>(pm_string_source(string)), pm_string_length(string));
```

File: test/prism_regression/regexp.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# typed: true
+
+/foo/
```
Comment (amomchilov) [inline test/prism_regression/regexp.rb:3]: Let's check for null issues w.r.t. empty:

```suggestion
// # empty Regexp

/foo/
```

(Idk why anybody would ever want an empty regex that matches everything, but 🤷‍♂️)

**Review by amomchilov — APPROVED** [review]

File: parser/prism/Translator.cc [hunk 3]
```diff
@@ -476,6 +476,37 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
...
+        case PM_REGULAR_EXPRESSION_NODE: {
+            auto regularExpressionNode = reinterpret_cast<pm_regular_expression_node *>(node);
+            pm_location_t *loc = &regularExpressionNode->base.location;
+            pm_location_t *closingLoc = &regularExpressionNode->closing_loc;
+
+            // Sorbet represents the regex content as a vector of string nodes
+            parser::NodeVec parts;
+
+            auto source = parser.extractString(&regularExpressionNode->unescaped);
+            if (source != "") {
```
Comment (egiurleo) [inline parser/prism/Translator.cc:488]: @amomchilov Does this seem like the most reasonable way to make this check? Sorbet does not expect `parts` to contain anything if the regex is empty.

File: parser/prism/Parser.cc [hunk 4]
```diff
@@ -2,6 +2,13 @@
 
 namespace sorbet::parser::Prism {
 
+enum class PmStringType {
+    PM_STRING_CONSTANT = 0,
+    PM_STRING_SHARED = 1,
+    PM_STRING_OWNED = 2,
+    PM_STRING_MAPPED = 3,
+};
+
```
Comment (amomchilov) [inline parser/prism/Parser.cc:11]: Good idea, but not needed anymore 🥳 

```suggestion
```

File: parser/prism/Translator.cc [hunk 5]
```diff
@@ -476,6 +476,37 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
...
+
+            std::string_view optString;
+
+            auto optStart = closingLoc->start + 1; // one character after the closing `/`
+            auto optEnd = closingLoc->end;
+            auto optLength = optEnd - optStart;
+
+            // Some regexps have options, like `/foo/i`
+            // Some have no options
+            if (optLength > 0) {
```
Comment (amomchilov) [inline parser/prism/Translator.cc:502]: Nit: wording is a bit awkward ("some have" implies "some have no")

```suggestion
            if (optLength > 0) { // Optional options like the `i` in `/foo/i`
```

**Review by amomchilov — APPROVED** [review]

**Pull request merged**